### ISSUE TITLE
Changes to value files in other repositories are ignored by the /manifests API endpoint [multi-source-apps]

### ISSUE THREAD

**Issue by Cr4mble** [argoproj/argo-cd#13628]
Checklist:

* [X] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [X] I've included steps to reproduce the bug.
* [X] I've pasted the output of `argocd version`.

**Describe the bug**

We want to implement the new feature "multi-source-app" and for that we are splitting up our current repository.
One repository will have the helm chart and the other repository will have the values files for all tenants.
```
spec:
  project: development
  sources:
    - repoURL: 'https://github.com/xxx/k8s-deployment'
      path: charts/chart_here
      targetRevision: dev
      helm:
        valueFiles:
          - values.yaml
          - $values/tenants/values.base.yaml
          - $values/tenants/values/app.tenant-xxx.yaml
        releaseName: test
        version: v3
    - repoURL: https://github.com/xxx/k8s-tenant-configurations'
      targetRevision: dev
      ref: values
```
If we now querying the API for the manifest of the app on the URL
`https://argo_url/api/v1/applications/NAMESPACE/manifests`
a 500 internal server error is returned.

error":"`helm template . --name-template test --namespace tenant-xxx --kube-version 1.25 --values .charts/chart_here/values.yaml --values .tenants/values.base.yaml --values .tenants/values/app.tenant-xxx.yaml .......... failed exit status 1: Error: open .tenants/values.base.yaml: no such file or directory

It seems that the helm template command is searching for the two values files in the current directory but as they are in another repository the command is not able to find them.

I deleted the system and build it from scratch. Then the error was gone but any changes
on the files values.base.yaml and app.tenant-xxx.yaml are ignored. The console shows the difference if i update the files but the API endpoint /manifests still shows the old values.
So we can't rely on the /manifests endpoint as it doesn't show the correct manifest.

We use the /manifests endpoint in the PreSync to get the image tag of our application to compare it with the current 
used image tag on k8s. If they differ then we are triggering a database backup.

**To Reproduce**

Use a multi source app with the helm chart in one repository and another values file in another repository. Then override an attribute in the values file of the second repository. Request the /manifests endpoint URL of the API and see that the override in the second values file is ignored. Although the Argo console shows the difference in the "App diff" view.

**Expected behavior**

I would expect that the /manifests endpoint of the API is returning the manifest of the application that is requested.

**Screenshots**

**Version**

```
{
    "Version": "v2.7.2+cbee7e6.dirty",
    "BuildDate": "2023-05-12T13:43:26Z",
    "GitCommit": "cbee7e6011407ed2d1066c482db74e97e0cc6bdb",
    "GitTreeState": "dirty",
    "GoVersion": "go1.19.6",
    "Compiler": "gc",
    "Platform": "linux/amd64",
    "KustomizeVersion": "v5.0.1 2023-03-14T01:32:48Z",
    "HelmVersion": "v3.11.2+g912ebc1",
    "KubectlVersion": "v0.24.2",
    "JsonnetVersion": "v0.19.1"
}
```

**Logs**

```
Paste any relevant application logs here.
```

**Comment by sanderma** [comment]
Yes I also have this problem, even better: If you run the example from [the doc](https://argo-cd.readthedocs.io/en/stable/user-guide/multiple_sources/#helm-value-files-from-external-git-repository) it also doesn't work.

Furthermore: Argo should resolve the `ref` variable in output, showing where it gets resources from. Given the example from the doc it now shows `$values/charts/prometheus/values.yaml` and should show `https://git.example.gom/org/value-files.git/charts/prometheus/values.yaml`
It's then easier to see if it's configured correctly.

**Comment by ishitasequeira** [comment]
The issue should be fixed in 2.11 as we have added support for multiple sources to manifests api endpoint. [#17804](https://github.com/argoproj/argo-cd/pull/17804)
